content-security-policy/nonce-hiding/script-nonces-hidden.tentative.html

```diff
@@ -114,6 +114,46 @@
     }, "createElement.setAttribute.");
 </script>
 
+<!-- Custom Element -->
+<script nonce="abc">
+  var eventList = [];
+  class NonceElement extends HTMLElement {
+    static get observedAttributes() {
+      return ['nonce'];
+    }
+
+    constructor() {
+      super();
+    }
+
+    attributeChangedCallback(name, oldValue, newValue) {
+      eventList.push({
+        type: "AttributeChanged",
+        name: name,
+        oldValue: oldValue,
+        newValue: newValue
+      });
+    }
+
+    connectedCallback() {
+      eventList.push({
+        type: "Connected",
+      });
+    }
+  }
+
+  customElements.define("nonce-element", NonceElement);
+</script>
+<nonce-element nonce="abc"></nonce-element>
+<script nonce="abc">
+  test(t => {
+    assert_equals(eventList.length, 3);
+    assert_object_equals(eventList[0], { type: "AttributeChanged", name: "nonce", oldValue: null, newValue: "abc" });
```

Comment:
So this means that for custom elements we would expose the nonce attribute value to script?

Reply:
Yes. The underlying PR does not intend to prevent script from accessing the nonce, but to protect it from other exfiltration mechanisms that don't require script.